wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -195,23 +196,30 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
     // Check for status change.
     if ($currentStatus !== $newStatus) {
       if ($newStatus === 'authorized') {
-        self::sendAuthorizationEmail($contactId);
+        self::sendAuthorizationEmail($contactId, $subType);
       }
       elseif ($newStatus === 'unauthorized') {
-        self::sendUnAuthorizationEmail($contactId);
+        self::sendUnAuthorizationEmail($contactId, $subType);
       }
     }
   }
 
   /**
    * Send Authorization Email to contact.
    */
-  private static function sendAuthorizationEmail($contactId) {
+  private static function sendAuthorizationEmail($contactId, $subType) {
     try {
+      // Determine the template based on dynamic subtype.
+      $templateId = $subType == 4 ? 78 : ($subType == 5 ? 83 : null);
```

Comment:
@pokhiii 
Sure, I'll note this down where we avoid using hardcoded IDs. For now, I've created constants for the template IDs and used those to reference the templates.

I have a question regarding the subType value. I've noticed that the subType returns values like 4 and 5 for "Collection Camp" and "Dropping Center." However, I couldn't change these values from the WordPress admin dashboard. I'm fetching these IDs dynamically from the objectRef with $subType = $objectRef['subtype'] ?? '';. When I log this, it returns the ID rather than the name.

I'm wondering if I should create constants for these subType values. It doesn't seem ideal since we're fetching the subType dynamically. Could you advise on how we can avoid using the ID for subType and instead use the name directly?